motionplan/plan.go

```diff
@@ -53,7 +53,7 @@ func OffsetPlan(plan Plan, offset spatialmath.Pose) Plan {
 	for _, step := range path {
 		newStep := make(PathStep, len(step))
 		for frame, pose := range step {
-			newStep[frame] = referenceframe.NewPoseInFrame(referenceframe.World, spatialmath.Compose(offset, pose.Pose()))
+			newStep[frame] = referenceframe.NewPoseInFrame(pose.Parent(), spatialmath.Compose(offset, pose.Pose()))
```

Comment:
hmmm, I am thinking about an example where our PiF might not be in the world frame..
Perhaps an example could be planning movement for a gripper with respect to an arm's end effector?? I'm really not sure..
Ultimately, I want to know if there is anything valid stopping us from using the provided framesystem within the planRequest to transform into the world frame.

Reply:
The PiF will not be in the world frame, if the user passed `NewSimplePlan` a Path that contains PiFs that are not in world frame. This is one example; there are many others. For example a user implementation of a `Plan`, as it is an interface.

Things like the builtin motion service are merely among the potential users of this function.